Use MT5 builtin’s translation system on all the modpack

louisroyer opened the pull request:
If you are using an already created map to test this PR, you may need to break and replace some nodes (like chests) to see effects.

louisroyer committed:
Use MT-5 translation system for concrete mod

louisroyer committed:
Use MT-5 translation system for extranodes mod

louisroyer committed:
Use MT-5 translation system for chests mod

louisroyer committed:
Use MT-5 translation system for cnc mod

louisroyer committed:
Use MT-5 translation system for worldgen mod

louisroyer committed:
Use MT-5 translation system for wrench mod

louisroyer committed:
Use MT-5 translation system for technic mod

louisroyer committed:
Remove translation strings from `minetest.log` output and from `print()` output
Since they are not translated anyway.

louisroyer committed:
Fix translation of iron

And preserve compatibility with [toolranks
mod](https://github.com/lisacvuk/minetest-toolranks/).

louisroyer committed:
Fix translation of wrought iron blocks

Note: nodes to convert from steel blocks to wroughtiron blocks now
require to be added manually since doing this automatically breaks
translations.

louisroyer commented on technic/init.lua:
```diff
@@ -48,6 +40,6 @@ dofile(modpath.."/tools/init.lua")
 
 if minetest.settings:get_bool("log_mods") then
-	print(S("[Technic] Loaded in %f seconds"):format(os.clock() - load_start))
+	print(string.format("[Technic] Loaded in %f seconds", os.clock() - load_start))
```
to fix

louisroyer commented on technic/machines/other/frames.lua:
```diff
@@ -298,10 +298,7 @@ for zp = 0, 1 do
-					.. " tried to place " .. itemstack:get_name()
-					.. " at protected position "
-					.. minetest.pos_to_string(pos))
+				minetest.log("action", placer:get_player_name().." tried to place "..itemstack:get_name().." at protected position "..minetest.pos_to_string(pos))
```
to fix

louisroyer commented on technic/machines/other/frames.lua:
```diff
@@ -327,10 +324,7 @@ for zp = 0, 1 do
-						.. " tried to place " .. itemstack:get_name()
-						.. " at protected position "
-						.. minetest.pos_to_string(pos))
+					minetest.log("action", placer:get_player_name().." tried to place "..itemstack:get_name().." at protected position "..minetest.pos_to_string(pos))
```
to fix

louisroyer commented on technic_chests/common.lua:
```diff
@@ -47,20 +50,14 @@ function technic.chests.inv_take(pos, listname, index, stack, player)
-	minetest.log("action", player:get_player_name()..
-		" moves stuff in chest at "
-		..minetest.pos_to_string(pos))
+	minetest.log("action", player:get_player_name().." moves stuff in chest at "..minetest.pos_to_string(pos))
```
to fix

louisroyer commented on technic_chests/common.lua:
```diff
@@ -47,20 +50,14 @@ function technic.chests.inv_take(pos, listname, index, stack, player)
-	minetest.log("action", player:get_player_name() ..
-			" moves " .. stack:get_name() ..
-			" to chest at " .. minetest.pos_to_string(pos))
+	minetest.log("action", player:get_player_name().." moves "..stack:get_name().." to chest at "..minetest.pos_to_string(pos))
```
to fix

louisroyer commented on technic_chests/common.lua:
```diff
@@ -47,20 +50,14 @@ function technic.chests.inv_take(pos, listname, index, stack, player)
-	minetest.log("action", player:get_player_name() ..
-			" takes " .. stack:get_name()  ..
-			" from chest at " .. minetest.pos_to_string(pos))
+	minetest.log("action", player:get_player_name().." takes "..stack:get_name().." from chest at "..minetest.pos_to_string(pos))
```
to fix

louisroyer reviewed (commented):
<s>needs to rework on some points</s>

louisroyer committed:
Fix crash when switching station is placed

louisroyer commented:
@louisroyer TODO: rebase and re-test

louisroyer committed:
Removes depends.txt

louisroyer committed:
Add min minetest version

louisroyer committed:
Add toolranks in luacheckrc && update to toolranks 1.2 api

louisroyer committed:
Fix luacheck syntax warnings

louisroyer committed:
Fix battery box tier localization

louisroyer commented:
Note: some strings on battery box nodes overflow the formspec (but it is not a regression since this also occurs on master). Must I fix the size of the formspec in this PR (by extending on the right) or do it in another one?
example:
![2020-08-12-214801_536x539_scrot](https://user-images.githubusercontent.com/55180044/90060804-8da14780-dce5-11ea-89ac-a8a5d320ab42.png)

auouymous commented:
Instead of increasing formspec width, those two lines could be moved below the upgrade slots and power level. The digiline channel button could be moved to the right or left side of the upgrade slots.

BuckarooBanzay commented:
> Instead of increasing formspec width, those two lines could be moved below the upgrade slots and power level. The digiline channel button could be moved to the right or left side of the upgrade slots.

can't this be done in another PR? This one is about translations only.

louisroyer commented:
> can't this be done in another PR? This one is about translations only.

ok, I created a new issue https://github.com/minetest-mods/technic/issues/562 and will fix this in another PR.

louisroyer committed:
Update locales

Using
[update_translations](https://github.com/minetest-tools/update_translations)
tool.
Also renamed translations files.

louisroyer committed:
Update documentation

SmallJoker commented:
Sorry, I didn't remember there was a PR to change the translation system.

Would you - at any chance - have time to rebase this PR? To reduce additional work, it might make sense to reorder and squash related commits using `git rebase -i HEAD~20`.

louisroyer commented:
I will try to have a look at this before the end of the month.

SmallJoker commented:
@louisroyer Would you like to continue on this PR sometime?

SmallJoker commented:
Abandoned. https://github.com/minetest-mods/technic/pull/664 is a follow-up PR.
